sympy/integrals/tests/test_prde.py

```diff
@@ -144,6 +146,41 @@ def test_prde_no_cancel():
     assert (Matrix([q])*V[0][:6, :])[0] == Poly(x - 1/2, t, domain='QQ(x)')
 
 
+def test_prde_cancel_liouvillian():
+    ### 1. case == 'primitive'
+    # used when integrating (log(-1/y)/2 - log(1/y)/2)/y - (log(1 - 1/y)/2 - log(1 + 1/y)/2)/y
+    # Not taken from 'the' book
+    DE = DifferentialExtension(extension={'D': [Poly(1, x), Poly(1/x, t)]})
+    assert prde_cancel_liouvillian(Poly(-1/(x - 1), t), [Poly(-x + 1, t), Poly(1, t)], 1, DE) == \
```

Comment:
Yes, (that is a wrong comment above, I wonder how I made the mistake, it took me sometime to come up with what this integral is):
```
[gxyd@stallman sympy{liouvillian}]$ bin/isympy
IPython console for SymPy 1.0.1.dev (Python 3.6.0-64-bit) (ground types: python)

These commands were executed:
>>> from __future__ import division
>>> from sympy import *
>>> x, y, z, t = symbols('x y z t')
>>> k, m, n = symbols('k m n', integer=True)
>>> f, g, h = symbols('f g h', cls=Function)
>>> init_printing()

Documentation can be found at http://docs.sympy.org/dev


In [1]: import pdb

In [2]: from sympy.integrals.risch import risch_integrate, DifferentialExtension

In [3]: pdb.run('risch_integrate(log(x) - log(x - 1), x)')
> <string>(1)<module>()
(Pdb) s
--Call--
> /home/gxyd/foss/sympy/sympy/core/basic.py(105)__hash__()
-> def __hash__(self):
(Pdb) b sympy/integrals/prde.py:448
Breakpoint 1 at /home/gxyd/foss/sympy/sympy/integrals/prde.py:448
(Pdb) c
> /home/gxyd/foss/sympy/sympy/integrals/prde.py(448)prde_cancel_liouvillian()
-> H = []
(Pdb) p b
Poly(-1/(x - 1), _t0, domain='ZZ(x)')
(Pdb) p Q
[Poly(-x + 1, _t0, domain='ZZ(x)'), Poly(1, _t0, domain='ZZ(x)')]
(Pdb) p n
1
(Pdb) p DE
DifferentialExtension(dict([('f', log(x) - log(x - 1)), ('x', x), ('T', [x, _t0, _t1]), ('D', [Poly(1, x, domain='ZZ'), Poly(1/x, _t0, domain='ZZ(x)'), Poly(1/(x - 1), _t1, domain='ZZ(x)')]), ('fa', Poly(-_t1 + _t0, _t1, domain='ZZ[_t0]')), ('fd', Poly(1, _t1, domain='ZZ')), ('Tfuncs', [Lambda(_i, log(_i)), Lambda(_i, log(_i - 1))]), ('backsubs', []), ('E_K', []), ('E_args', []), ('L_K', [1, 2]), ('L_args', [x, x - 1]), ('cases', ['base', 'primitive', 'primitive']), ('case', 'primitive'), ('t', _t0), ('d', Poly(1/x, _t0, domain='ZZ(x)')), ('newf', _t0 - _t1), ('level', -2), ('dummy', True)]))
(Pdb) p DE.D
[Poly(1, x, domain='ZZ'), Poly(1/x, _t0, domain='ZZ(x)'), Poly(1/(x - 1), _t1, domain='ZZ(x)')]
(Pdb) p DE.level
-2
(Pdb) 
```